go-controller/pkg/cni/cniserver.go

```diff
@@ -46,7 +48,11 @@ import (
 // started.
 
 // NewCNIServer creates and returns a new Server object which will listen on a socket in the given path
-func NewCNIServer(rundir string, factory factory.NodeWatchFactory) *Server {
+func NewCNIServer(rundir string, factory factory.NodeWatchFactory, kclient kubernetes.Interface, mode string) (*Server, error) {
+	if mode == types.NodeModeSmartNIC {
```

Comment:
you will never hit this code? do we need this check ...

see 
	if config.OvnKubeNode.Mode == types.NodeModeSmartNIC {
	err = cniServer.Start(cni.HandleCNIRequest)			n.watchSmartNicPods()
	} else {
		// start the cni server
		kclient, ok := n.Kube.(*kube.Kube)
		if !ok {
			return fmt.Errorf("cannot get kubeclient for starting CNI server")
		}
		var cniServer *cni.Server
		cniServer, err = cni.NewCNIServer("", n.watchFactory, kclient.KClient, config.OvnKubeNode.Mode)
		if err == nil {
			err = cniServer.Start(cni.HandleCNIRequest)
		}
	}

Reply:
Yes i am aware.

Should we just rely on the consumer of cniserver to provide valid mode ? 
or should cniserver also validate (as if we got here, someone did something unexpected and we should fail)?
in the latter case perhaps i should make the validation more strict (as only "full" and "smar-nic-host" are supported in case additional modes are added down the road.)

Reply:
I see, I missed the part that you need the mode for the NewPodLister

Reply:
Originally i didnt want the `NewCNIServer()` method to rely on config for object creation. seemed to me like the caller will have a better understands towards what he is going to get from `NewCNIServer()` if this parameter is passed to this function.

However if you feel its simpler to just use config directly im fine with changing it.